EIPS/eip-1185.md

```diff
@@ -31,50 +34,239 @@ The DNS resolver interface consists of two functions to set DNS information and
+     * @param node the namehash of the node for which to check the records
+     * @param name the namehash of the node for which to check the records
+     */
+    function hasDNSRecords(bytes32 node, bytes32 name) public view returns (bool) {
+        return (nameEntriesCount[node][versions[node]][name] != 0);
+    }
+
+    /**
+     * Clear all information for a DNS zone.
+     * @param node the namehash of the node for which to clear the zone
+     */
+    function clearDNSZone(bytes32 node) public authorised(node) {
+        versions[node]++;
+        emit DNSZoneCleared(node);
+    }
+
+    /**
+     * setZonehash sets the hash for the zone.
+     * May only be called by the owner of that node in the ENS registry.
+     * @param node The node to update.
+     * @param hash The zonehash to set
+     */
+    function setZonehash(bytes32 node, bytes calldata hash) external authorised(node) {
+        bytes memory oldhash = zonehashes[node];
+        zonehashes[node] = hash;
+        emit DNSZonehashChanged(node, oldhash, hash);
+    }
+
+    /**
+     * zonehash obtains the hash for the zone.
+     * @param node The ENS node to query.
+     * @return The associated contenthash.
+     */
+    function zonehash(bytes32 node) external view returns (bytes memory) {
+        return zonehashes[node];
+    }
+
+    function supportsInterface(bytes4 interfaceID) virtual override public pure returns(bool) {
+        return interfaceID == DNS_RECORD_INTERFACE_ID ||
+               interfaceID == DNS_ZONE_INTERFACE_ID ||
+               super.supportsInterface(interfaceID);
+    }
+
+    function setDNSRRSet(
+        bytes32 node,
+        bytes memory name,
+        uint16 resource,
+        bytes memory data,
+        uint256 offset,
+        uint256 size,
+        bool deleteRecord) private
+    {
+        uint256 version = versions[node];
+        bytes32 nameHash = keccak256(name);
+        bytes memory rrData = data.substring(offset, size);
+        if (deleteRecord) {
+            if (records[node][version][nameHash][resource].length != 0) {
+                nameEntriesCount[node][version][nameHash]--;
+            }
+            delete(records[node][version][nameHash][resource]);
+            emit DNSRecordDeleted(node, name, resource);
+        } else {
+            if (records[node][version][nameHash][resource].length == 0) {
+                nameEntriesCount[node][version][nameHash]++;
+            }
+            records[node][version][nameHash][resource] = rrData;
+            emit DNSRecordChanged(node, name, resource, rrData);
+        }
+    }
+}
+```
+
+## Security Considerations
+
+Security of this solution would be dependent on security of the records within the ENS domain.  This degenenrates to the security of the key(s) which have authority over that domain.
 
 ## Copyright
-Copyright and related rights waived via [CC0](../LICENSE.md).
+
+Copyright and related rights waived via [CC0](/LICENSE)
```

Comment:
To make links work consistently between eips.ethereum.org and GitHub, you need to use a URL relative to the current file (so not starting with a `/`) and include the `.md` extension.

```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```